pandas/tests/frame/test_query_eval.py

```diff
@@ -695,6 +695,23 @@ def test_inf(self):
             result = df.query(q, engine=self.engine, parser=self.parser)
             tm.assert_frame_equal(result, expected)
 
+    def test_check_tz_aware_index_query(self, tz_aware_fixture):
+        # https://github.com/pandas-dev/pandas/issues/29463
+        tz = tz_aware_fixture
+        expected = pd.DataFrame(
+            index=pd.date_range(
+                start="2019-01-01", freq="1d", periods=10, tz=tz, name="time"
+            )
+        )
+
+        df = pd.DataFrame(
+            index=pd.date_range(
+                start="2019-01-01", freq="1d", periods=10, tz=tz, name="time"
+            )
+        )
+        result = df.query('"2018-01-03 00:00:00+00" < time')
```

Comment:
Like this, `time` is always greater than `2018-01-03 00:00:00+00` - I think @mroeschke was suggesting to use, say, `2019-01-04` in `expected`, `2019-01-01` in `df`, and then `2019-01-03` in `query`

Reply:
@MarcoGorelli  Thank you for the comment / appreciation.

If we use 2019-01-03 in query it results in earlier problem , because the shape of "expected" and "df" remains the same also their tz is changing , but in query the tz remains constant and when we query it to diff timezone their shape becomes inconsistent as the query tz  is always constant. (So it always result in +1 or -1 records and hence fails.)

Hence I think he suggested to use 2018-01-03 , as it will always result in consistent shape while checking tz-aware for all time zones.

![Screenshot 2020-05-07 at 02 01 14](https://user-images.githubusercontent.com/6196315/81225414-d078c280-9006-11ea-9979-ab6bf91a0353.png)

Reply:
Ah sorry I'd misunderstood, thanks!

Reply:
In that case, it seems that your index for both `df` and `expected` are identical - if so, can you make a variable to share them?

Reply:
Done ! Thanks.